### DISCUSSION TITLE
PR算法用的是哪个算法呀

### DISCUSSION THREAD

**Discussion by TerrenceYi** [alibaba/GraphScope#3126] [Q&A] [answered]
Graphscope里的PR算法对出度为0的顶点处理似乎不同于谷歌算法，具体是使用了哪个算法呀

**Comment by TerrenceYi** [comment]
请问这个PageRank里面的base是怎么来的呀，似乎是一个动态调整的值，不同于谷歌PageRank中的1/N，另外，这个算法会最终收敛并且让收敛后的网页排名之和为1吗
求好心人回答

**Comment by mengke-mk** [comment, marked as answer]
We implement the PR algorithm followed the [LDBC spec, page 32](https://arxiv.org/pdf/2011.15028.pdf), which handles dangling nodes.
FYI. https://github.com/alibaba/libgrape-lite/blob/master/examples/analytical_apps/pagerank/pagerank_directed.h#L64

**Reply by TerrenceYi** [reply]
Thank you very much,but I was looking at the code in Graphscope-main, unlike the code in libgrape, which assigns base to vertices of degree 0。
